syntax = "proto3";
package BankService;

service BankService {
    rpc MsgDelivery (MsgDeliveryRequest) returns (MsgDeliveryResponse) {}
}

enum Interface {
    QUERY = 0;
    DEPOSIT = 1;
    WITHDRAW = 2;
}

enum Status {
    SUCCESS = 0;
    FAILURE = 1;
    ERROR = 2;
}

message MsgDeliveryRequest {
    int64 s_id = 1;
    int64 d_id = 2;
    Interface interface = 3;
    float amount = 4;
}

message MsgDeliveryResponse {
    int64 id = 1;
    Status responseStatus = 2;
    float amount = 3;
}